src/v2/Components/__tests__/Timer.jest.tsx

```diff
@@ -11,58 +11,143 @@ require("v2/Utils/getCurrentTimeAsIsoString").__setCurrentTime(
+    expect(getTimerText(timer)).toMatch("01m")
+    expect(getTimerText(timer)).toMatch("59s")
+
+    // Thursday, May 10, 2018 8:42:32.000 PM UTC
+    timer = renderer.create(<Timer endDate="2018-05-10T20:42:32+00:00" />)
+    expect(getTimerText(timer)).toMatch("00d")
+    expect(getTimerText(timer)).toMatch("00h")
+    expect(getTimerText(timer)).toMatch("20m")
+    expect(getTimerText(timer)).toMatch("00s")
+
+    // Thursday, May 10, 2018 8:22:42.000 PM UTC
+    timer = renderer.create(<Timer endDate="2018-05-10T20:22:42+00:00" />)
+    expect(getTimerText(timer)).toMatch("00d")
+    expect(getTimerText(timer)).toMatch("00h")
+    expect(getTimerText(timer)).toMatch("0m")
+    expect(getTimerText(timer)).toMatch("10s")
+
+    // In the past
+    timer = renderer.create(<Timer endDate="2018-04-10T20:22:42+00:00" />)
+    expect(getTimerText(timer)).toMatch("00d")
+    expect(getTimerText(timer)).toMatch("00h")
+    expect(getTimerText(timer)).toMatch("00m")
+    expect(getTimerText(timer)).toMatch("00s")
+  })
+
+  it("counts down to zero", () => {
+    let timer = renderer.create(<Timer endDate="2018-05-14T10:23:10+00:00" />)
+    expect(getTimerText(timer)).toMatch("03d")
+    expect(getTimerText(timer)).toMatch("14h")
+    expect(getTimerText(timer)).toMatch("00m")
+    expect(getTimerText(timer)).toMatch("38s")
+
+    timer = renderer.create(<Timer endDate="2018-05-14T10:23:08+00:00" />)
+    jest.runOnlyPendingTimers()
+    expect(getTimerText(timer)).toMatch("03d")
+    expect(getTimerText(timer)).toMatch("14h")
+    expect(getTimerText(timer)).toMatch("00m")
+    expect(getTimerText(timer)).toMatch("36s")
+
+    timer = renderer.create(<Timer endDate="2018-05-14T10:22:08+00:00" />)
+    jest.runOnlyPendingTimers()
+    expect(getTimerText(timer)).toMatch("03d")
+    expect(getTimerText(timer)).toMatch("13h")
+    expect(getTimerText(timer)).toMatch("59m")
+    expect(getTimerText(timer)).toMatch("36s")
+  })
 })
 
-it("counts down to zero", () => {
-  let timer = renderer.create(<Timer endDate="2018-05-14T10:23:10+00:00" />)
-  expect(getTimerText(timer)).toMatch("03d")
-  expect(getTimerText(timer)).toMatch("14h")
-  expect(getTimerText(timer)).toMatch("00m")
-  expect(getTimerText(timer)).toMatch("38s")
-
-  timer = renderer.create(<Timer endDate="2018-05-14T10:23:08+00:00" />)
-  jest.runOnlyPendingTimers()
-  expect(getTimerText(timer)).toMatch("03d")
-  expect(getTimerText(timer)).toMatch("14h")
-  expect(getTimerText(timer)).toMatch("00m")
-  expect(getTimerText(timer)).toMatch("36s")
-
-  timer = renderer.create(<Timer endDate="2018-05-14T10:22:08+00:00" />)
-  jest.runOnlyPendingTimers()
-  expect(getTimerText(timer)).toMatch("03d")
-  expect(getTimerText(timer)).toMatch("13h")
-  expect(getTimerText(timer)).toMatch("59m")
-  expect(getTimerText(timer)).toMatch("36s")
+describe("when the start date and the end dates are set", () => {
+  it("formats the remaining time in '00d  00h  00m  00s'", () => {
```

Comment:
This isn't super descriptive, it should probably mention something about the new behavior (that the resulting timer returns formatted time until the start).